File: dataset/output/PERPLEXITY_llama-3-sonar-large-32k-chat/Task34_PERPLEXITY_llama-3-sonar-large-32k-chat.java
public class Narcissistic {
    public static boolean isNarcissistic(int n) {
        int sum = 0, temp = n;
        int digits = String.valueOf(n).length();
        while (temp != 0) {
            int digit = temp % 10;
            sum += Math.pow(digit, digits);
            temp /= 10;
        }
        return sum == n;
    }
    public static void main(String[] args) {
        System.out.println(isNarcissistic(153)); // true
        System.out.println(isNarcissistic(1652)); // false
    }
}

File: dataset/output/PERPLEXITY_llama-3-sonar-large-32k-chat/Task34_PERPLEXITY_llama-3-sonar-large-32k-chat.cpp
#include <cmath>
#include <string>

bool narcissistic(int value) {
    int sum = 0;
    std::string strValue = std::to_string(value);
    for (char digit : strValue) {
        sum += pow(std::stoi(std::string(1, digit)), strValue.length());
    }
    return value == sum;
}